### ISSUE TITLE
[Fabric] Server crash when steam boiler heat source is updated

### ISSUE THREAD

**Issue by ajh123** [mrh0/createaddition#407]
**Describe the bug**
The server crashes when vanilla soul campfire is broken.

**To Reproduce**
* I have set up a simple steam boiler / engine.
* The heat source comes from the soul campfire from the base game.
~~* I put a blaze burner in my inventory (at least the hotbar's first slot).~~
~~* I then break a single campfire with an empty hand.~~
* The heat source is updated (breaking or placing a new one)
* My server then crashes.

**Additional context**
This _might_ be a bug caused by Create (or another different mod in general), but create addition shows up in the logs.

**Versions**

- Mod version: `createaddition+1.19.2-1.19.2-20221128a`
- Fabric version: 
   * Loader: `0.14.10-1.19.2`
   * Fabric API: `fabric-api-0.67.1+1.19.2`

- Create version: `create-fabric-0.5.0g-796+1.19.2`

Yes, those were just copied from the jars.

**The crash report:**
```java
---- Minecraft Crash Report ----
// Surprise! Haha. Well, this is awkward.

Time: 2022-12-06 14:40:47
Description: Ticking block entity

java.lang.IllegalArgumentException: Cannot get property class_2754{name=blaze, clazz=class com.simibubi.create.content.contraptions.processing.burner.BlazeBurnerBlock$HeatLevel, values=[NONE, SMOULDERING, FADING, KINDLED, SEETHING]} as it does not exist in Block{minecraft:air}
	at net.minecraft.class_2688.method_11654(class_2688.java:98)
	at com.mrh0.createaddition.CreateAddition.lambda$onInitialize$0(CreateAddition.java:41)
	at com.mrh0.createaddition.CreateAddition$$Lambda$5632/0x000000004e63dad0.getActiveHeat(Unknown Source)
	at com.simibubi.create.content.contraptions.fluids.tank.BoilerHeaters.getActiveHeat(BoilerHeaters.java:46)
	at com.simibubi.create.content.contraptions.fluids.tank.BoilerData.updateTemperature(BoilerData.java:343)
	at com.simibubi.create.content.contraptions.fluids.tank.BoilerData.tick(BoilerData.java:84)
	at com.simibubi.create.content.contraptions.fluids.tank.FluidTankTileEntity.tick(FluidTankTileEntity.java:112)
	at com.simibubi.create.foundation.tileEntity.SmartTileEntityTicker.tick(SmartTileEntityTicker.java:15)
	at net.minecraft.class_2818$class_5563.redirect$zip000$checkProcessTEs(class_2818.java:784)
	at net.minecraft.class_2818$class_5563.method_31703(class_2818.java:660)
	at net.minecraft.class_2818$class_5564.method_31703(class_2818.java:714)
	at net.minecraft.class_1937.method_18471(class_1937.java:456)
	at net.minecraft.class_3218.method_18765(class_3218.java:376)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:875)
	at net.minecraft.class_3176.method_3813(class_3176.java:289)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:819)
	at net.minecraft.server.MinecraftServer.handler$zim000$modifiedRunLoop(MinecraftServer.java:3885)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:644)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257)
	at net.minecraft.server.MinecraftServer$$Lambda$10497/0x0000000050474060.run(Unknown Source)
	at java.base/java.lang.Thread.run(Thread.java:884)


A detailed walkthrough of the error, its code path and all known details is as follows:
---------------------------------------------------------------------------------------

-- Head --
Thread: Server thread
Stacktrace:
	at net.minecraft.class_2688.method_11654(class_2688.java:98)
	at com.mrh0.createaddition.CreateAddition.lambda$onInitialize$0(CreateAddition.java:41)
	at com.mrh0.createaddition.CreateAddition$$Lambda$5632/0x000000004e63dad0.getActiveHeat(Unknown Source)
	at com.simibubi.create.content.contraptions.fluids.tank.BoilerHeaters.getActiveHeat(BoilerHeaters.java:46)
	at com.simibubi.create.content.contraptions.fluids.tank.BoilerData.updateTemperature(BoilerData.java:343)
	at com.simibubi.create.content.contraptions.fluids.tank.BoilerData.tick(BoilerData.java:84)
	at com.simibubi.create.content.contraptions.fluids.tank.FluidTankTileEntity.tick(FluidTankTileEntity.java:112)
	at com.simibubi.create.foundation.tileEntity.SmartTileEntityTicker.tick(SmartTileEntityTicker.java:15)
	at net.minecraft.class_2818$class_5563.redirect$zip000$checkProcessTEs(class_2818.java:784)
	at net.minecraft.class_2818$class_5563.method_31703(class_2818.java:660)
	at net.minecraft.class_2818$class_5564.method_31703(class_2818.java:714)
	at net.minecraft.class_1937.method_18471(class_1937.java:456)
	at net.minecraft.class_3218.method_18765(class_3218.java:376)

-- Block entity being ticked --
Details:
	Name: create:fluid_tank // com.simibubi.create.content.contraptions.fluids.tank.FluidTankTileEntity
	Block: Block{create:fluid_tank}[bottom=true,light_level=0,shape=plain,top=false]
	Block location: World: (100,69,81), Section: (at 4,5,1 in 6,4,5; chunk contains blocks 96,-64,80 to 111,319,95), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
	Block: Block{create:fluid_tank}[bottom=true,light_level=0,shape=plain,top=false]
	Block location: World: (100,69,81), Section: (at 4,5,1 in 6,4,5; chunk contains blocks 96,-64,80 to 111,319,95), Region: (0,0; contains chunks 0,0 to 31,31, blocks 0,-64,0 to 511,319,511)
Stacktrace:
	at net.minecraft.class_2818$class_5563.method_31703(class_2818.java:660)
	at net.minecraft.class_2818$class_5564.method_31703(class_2818.java:714)
	at net.minecraft.class_1937.method_18471(class_1937.java:456)
	at net.minecraft.class_3218.method_18765(class_3218.java:376)
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:875)
	at net.minecraft.class_3176.method_3813(class_3176.java:289)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:819)
	at net.minecraft.server.MinecraftServer.handler$zim000$modifiedRunLoop(MinecraftServer.java:3885)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:644)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257)
	at net.minecraft.server.MinecraftServer$$Lambda$10497/0x0000000050474060.run(Unknown Source)
	at java.base/java.lang.Thread.run(Thread.java:884)

-- Affected level --
Details:
	All players: 1 total; [class_3222['samuelh2005'/253, l='ServerLevel[City]', x=99.08, y=67.22, z=84.97]]
	Chunk stats: 5394
	Level dimension: minecraft:overworld
	Level spawn location: World: (-32,70,32), Section: (at 0,6,0 in -2,4,2; chunk contains blocks -32,-64,32 to -17,319,47), Region: (-1,0; contains chunks -32,0 to -1,31, blocks -512,-64,0 to -1,319,511)
	Level time: 32702249 game time, 17495549 day time
	Level name: City
	Level game mode: Game mode: survival (ID 0). Hardcore: false. Cheats: true
	Level weather: Rain time: 1 (now: false), thunder time: 1 (now: false)
	Known server brands: fabric
	Level was modded: true
	Level storage version: 0x04ABD - Anvil
Stacktrace:
	at net.minecraft.server.MinecraftServer.method_3813(MinecraftServer.java:875)
	at net.minecraft.class_3176.method_3813(class_3176.java:289)
	at net.minecraft.server.MinecraftServer.method_3748(MinecraftServer.java:819)
	at net.minecraft.server.MinecraftServer.handler$zim000$modifiedRunLoop(MinecraftServer.java:3885)
	at net.minecraft.server.MinecraftServer.method_29741(MinecraftServer.java:644)
	at net.minecraft.server.MinecraftServer.method_29739(MinecraftServer.java:257)
	at net.minecraft.server.MinecraftServer$$Lambda$10497/0x0000000050474060.run(Unknown Source)
	at java.base/java.lang.Thread.run(Thread.java:884)

-- System Details --
Details:
	Minecraft Version: 1.19.2
	Minecraft Version ID: 1.19.2
	Operating System: Windows 10 (amd64) version 10.0
	Java Version: 17.0.1, IBM Corporation
	Java VM Version: Eclipse OpenJ9 VM (JRE 17 Windows 10 amd64-64-Bit Compressed References 20211207_55 (JIT enabled, AOT enabled)
OpenJ9   - 7d055dfcb
OMR      - e30892e2b
JCL      - fc67fbe50a0 based on jdk-17.0.1+12), Eclipse OpenJ9
	Memory: 402270000 bytes (383 MiB) / 2663383040 bytes (2540 MiB) up to 8589934592 bytes (8192 MiB)
	CPUs: 12
	Processor Vendor: GenuineIntel
	Processor Name: Intel(R) Core(TM) i5-10400 CPU @ 2.90GHz
	Identifier: Intel64 Family 6 Model 165 Stepping 3
	Microarchitecture: unknown
	Frequency (GHz): 2.90
	Number of physical packages: 1
	Number of physical CPUs: 6
	Number of logical CPUs: 12
	Graphics card #0 name: NVIDIA GeForce GT 1030 <-- This card is always identified wrong.
	Graphics card #0 vendor: NVIDIA (0x10de)      <-- This card is always identified wrong.
	Graphics card #0 VRAM (MB): 2048.00
	Graphics card #0 deviceId: 0x1d01
	Graphics card #0 versionInfo: DriverVersion=31.0.15.1694
	Memory slot #0 capacity (MB): 8192.00
	Memory slot #0 clockSpeed (GHz): 2.67
	Memory slot #0 type: DDR4
	Memory slot #1 capacity (MB): 8192.00
	Memory slot #1 clockSpeed (GHz): 2.67
	Memory slot #1 type: DDR4
	Virtual memory max (MB): 33717.07
	Virtual memory used (MB): 16264.44
	Swap memory total (MB): 17408.00
	Swap memory used (MB): 465.53
	JVM Flags: 4 total; -Xoptionsfile=C:\Users\samro\.jdks\semeru-17.0.1\lib\options.default -Xlockword:mode=default,noLockword=java/lang/String,noLockword=java/util/MapEntry,noLockword=java/util/HashMap$Entry,noLockword=org/apache/harmony/luni/util/ModifiedMap$Entry,noLockword=java/util/Hashtable$Entry,noLockword=java/lang/invoke/MethodType,noLockword=java/lang/invoke/MethodHandle,noLockword=java/lang/invoke/CollectHandle,noLockword=java/lang/invoke/ConstructorHandle,noLockword=java/lang/invoke/ConvertHandle,noLockword=java/lang/invoke/ArgumentConversionHandle,noLockword=java/lang/invoke/AsTypeHandle,noLockword=java/lang/invoke/ExplicitCastHandle,noLockword=java/lang/invoke/FilterReturnHandle,noLockword=java/lang/invoke/DirectHandle,noLockword=java/lang/invoke/ReceiverBoundHandle,noLockword=java/lang/invoke/DynamicInvokerHandle,noLockword=java/lang/invoke/FieldHandle,noLockword=java/lang/invoke/FieldGetterHandle,noLockword=java/lang/invoke/FieldSetterHandle,noLockword=java/lang/invoke/StaticFieldGetterHandle,noLockword=java/lang/invoke/StaticFieldSetterHandle,noLockword=java/lang/invoke/IndirectHandle,noLockword=java/lang/invoke/InterfaceHandle,noLockword=java/lang/invoke/VirtualHandle,noLockword=java/lang/invoke/PrimitiveHandle,noLockword=java/lang/invoke/InvokeExactHandle,noLockword=java/lang/invoke/InvokeGenericHandle,noLockword=java/lang/invoke/VarargsCollectorHandle,noLockword=java/lang/invoke/ThunkTuple -Xjcl:jclse29 -Xmx8G
	Fabric Mods: 
		architectury: Architectury 6.2.43
		automobility: Automobility 0.3+1.19.2
			advanced_runtime_resource_pack: Runtime Resource Pack 0.6.2
			jsonem: JsonEM 0.1.2+1.19
		carpet: Carpet Mod 1.4.83
		computercraft: CC: Restitched 1.101.2
			io_netty_netty-codec-http: netty-codec-http 4.1.77.Final
			org_squiddev_cobalt: Cobalt 0.5.5
		coordfinder: Coord Finder 1.19.2-1.0.1
		create: Create 0.5.0g-796+1.19.2
			com_google_code_findbugs_jsr305: jsr305 3.0.2
			fake-player-api: Fake Player API 0.5.0
			milk: Milk Lib 1.0.51
				dripstone_fluid_lib: Dripstone Fluid Lib 2.0.1
			porting_lib: Porting Lib 2.1.623+1.19.2
				forge_tags: Porting Lib Tags 3.0
				porting_lib_base: Porting Lib Base 2.1.623+1.19.2
					com_github_llamalad7_mixinextras: MixinExtras 0.1.1
					serialization_hooks: Serialization Hooks 0.3.26
				porting_lib_entity: Porting Lib Entity 2.1.623+1.19.2
				porting_lib_lazy_registration: Porting Lib Lazy Register 2.1.623+1.19.2
				porting_lib_model_generators: Porting Lib Model Generators 2.1.623+1.19.2
				porting_lib_models: Porting Lib Models 2.1.623+1.19.2
				porting_lib_networking: Porting Lib Networking 2.1.623+1.19.2
				porting_lib_obj_loader: Porting Lib Obj Loader 2.1.623+1.19.2
					porting_lib_model_loader: Porting Lib Model Loader 2.1.623+1.19.2
				porting_lib_transfer: Porting Lib Transfer 2.1.623+1.19.2
					porting_lib_extensions: Porting Lib Extensions 2.1.623+1.19.2
						porting_lib_accessors: Porting Lib Accessors 2.1.623+1.19.2
						porting_lib_attributes: Porting Lib Attributes 2.1.623+1.19.2
							porting_lib_constants: Porting Lib Constants 2.1.623+1.19.2
						porting_lib_common: Porting Lib Common 2.1.623+1.19.2
			reach-entity-attributes: Reach Entity Attributes 2.3.0
			registrate-fabric: Registrate for Fabric 1.1.44-MC1.19.2
		createaddition: Create Crafts & Additions ${version}
			team_reborn_energy: Energy 2.2.0
		diagonalfences: Diagonal Fences 4.0.0
		dynmap: Dynmap 3.4
		exlinefurniture: Exline's Furniture 2.3.0
		fabric-api: Fabric API 0.67.1+1.19.2
			fabric-api-base: Fabric API Base 0.4.13+0d0f210290
			fabric-api-lookup-api-v1: Fabric API Lookup API (v1) 1.6.12+93d8cb8290
			fabric-biome-api-v1: Fabric Biome API (v1) 9.0.20+0d0f210290
			fabric-block-api-v1: Fabric Block API (v1) 1.0.1+12bfe4ea90
			fabric-command-api-v1: Fabric Command API (v1) 1.2.13+f71b366f90
			fabric-command-api-v2: Fabric Command API (v2) 2.1.9+93d8cb8290
			fabric-commands-v0: Fabric Commands (v0) 0.2.30+df3654b390
			fabric-containers-v0: Fabric Containers (v0) 0.1.37+df3654b390
			fabric-content-registries-v0: Fabric Content Registries (v0) 3.4.3+6064db2a90
			fabric-convention-tags-v1: Fabric Convention Tags 1.1.3+93d8cb8290
			fabric-crash-report-info-v1: Fabric Crash Report Info (v1) 0.2.7+aeb40ebe90
			fabric-data-generation-api-v1: Fabric Data Generation API (v1) 5.3.1+ad31814d90
			fabric-dimensions-v1: Fabric Dimensions API (v1) 2.1.33+0d0f210290
			fabric-entity-events-v1: Fabric Entity Events (v1) 1.5.2+9244241690
			fabric-events-interaction-v0: Fabric Events Interaction (v0) 0.4.30+c6af733c90
			fabric-events-lifecycle-v0: Fabric Events Lifecycle (v0) 0.2.31+df3654b390
			fabric-game-rule-api-v1: Fabric Game Rule API (v1) 1.0.23+c6af733c90
			fabric-item-api-v1: Fabric Item API (v1) 1.6.1+fa140d5990
			fabric-item-groups-v0: Fabric Item Groups (v0) 0.3.35+9244241690
			fabric-lifecycle-events-v1: Fabric Lifecycle Events (v1) 2.2.2+9244241690
			fabric-loot-api-v2: Fabric Loot API (v2) 1.1.9+83a8659290
			fabric-loot-tables-v1: Fabric Loot Tables (v1) 1.1.12+9e7660c690
			fabric-message-api-v1: Fabric Message API (v1) 5.0.5+93d8cb8290
			fabric-mining-level-api-v1: Fabric Mining Level API (v1) 2.1.20+33fbc73890
			fabric-networking-api-v1: Fabric Networking API (v1) 1.2.7+9244241690
			fabric-networking-v0: Fabric Networking (v0) 0.3.24+df3654b390
			fabric-object-builder-api-v1: Fabric Object Builder API (v1) 4.1.2+06e0e25190
			fabric-particles-v1: Fabric Particles (v1) 1.0.12+79adfe0a90
			fabric-registry-sync-v0: Fabric Registry Sync (v0) 0.9.28+9244241690
			fabric-rendering-data-attachment-v1: Fabric Rendering Data Attachment (v1) 0.3.17+0d0f210290
			fabric-rendering-fluids-v1: Fabric Rendering Fluids (v1) 3.0.9+c6af733c90
			fabric-resource-conditions-api-v1: Fabric Resource Conditions API (v1) 2.1.1+aae9039d90
			fabric-resource-loader-v0: Fabric Resource Loader (v0) 0.8.1+0d0f210290
			fabric-screen-handler-api-v1: Fabric Screen Handler API (v1) 1.3.3+1cc24b1b90
			fabric-transfer-api-v1: Fabric Transfer API (v1) 2.1.3+9d081d6490
			fabric-transitive-access-wideners-v1: Fabric Transitive Access Wideners (v1) 1.3.2+42d99c3290
		fabric-tree-chopper: Fabric Tree Chopper 0.8.7
			cloth-config: Cloth Config v6 7.0.69
				cloth-basic-math: cloth-basic-math 0.6.1
			fabric-language-kotlin: Fabric Language Kotlin 1.7.4+kotlin.1.6.21
				org_jetbrains_kotlin_kotlin-reflect: kotlin-reflect 1.6.21
				org_jetbrains_kotlin_kotlin-stdlib: kotlin-stdlib 1.6.21
				org_jetbrains_kotlin_kotlin-stdlib-jdk7: kotlin-stdlib-jdk7 1.6.21
				org_jetbrains_kotlin_kotlin-stdlib-jdk8: kotlin-stdlib-jdk8 1.6.21
				org_jetbrains_kotlinx_kotlinx-coroutines-core-jvm: kotlinx-coroutines-core-jvm 1.6.1
				org_jetbrains_kotlinx_kotlinx-coroutines-jdk8: kotlinx-coroutines-jdk8 1.6.1
				org_jetbrains_kotlinx_kotlinx-serialization-cbor-jvm: kotlinx-serialization-cbor-jvm 1.3.2
				org_jetbrains_kotlinx_kotlinx-serialization-core-jvm: kotlinx-serialization-core-jvm 1.3.2
				org_jetbrains_kotlinx_kotlinx-serialization-json-jvm: kotlinx-serialization-json-jvm 1.3.2
			fiber: fiber 0.23.0-2
		fabricloader: Fabric Loader 0.14.10
		forgeconfigapiport: Forge Config API Port 4.2.6
			com_electronwill_night-config_core: core 3.6.5
			com_electronwill_night-config_toml: toml 3.6.5
		globaldataandresourcepacks: Global Data- & Resourcepacks 1.14.2_fabric
		java: Eclipse OpenJ9 VM 17
		krypton: Krypton 0.2.1
			com_velocitypowered_velocity-native: velocity-native 3.1.2-SNAPSHOT
		lithium: Lithium 0.10.2
		londonunderground: MTR London Underground Addon 1.19.2-3.2.0-1.4
		minecraft: Minecraft 1.19.2
		modern_industrialization: Modern Industrialization 1.3.2
			magna: Magna 1.8.0-1.19
				omega-config: OmegaConfig 1.0.8
			patchouli: Patchouli 1.19-70-FABRIC-SNAPSHOT
			playerabilitylib: Pal 1.6.0
		mtr: Minecraft Transit Railway 1.19.2-3.1.9
		mysql-jdbc: MySQL JDBC 8.0.28+20220312
		openpartiesandclaims: Open Parties and Claims 0.13.0
		puzzleslib: Puzzles Lib 4.1.8
		roughlyenoughitems: Roughly Enough Items 9.1.530
			error_notifier: Error Notifier 1.0.9
		shulkerboxtooltip: Shulker Box Tooltip 3.0.9-beta.1+1.19-rc2
			mm: Manningham Mills 2.3
		simplechunkloader: Simple Chunkloader 1.0.3
			libgui: LibGui 6.0.0-beta.5+1.19-pre1
				jankson: Jankson 4.1.1+j1.2.1
					blue_endless_jankson: jankson 1.2.1
				libninepatch: LibNinePatch 1.1.0
		smallships: Small Ships 2.0.0a1.1.1
		starlight: Starlight 1.1.1+fabric.ae22326
		status: Status 1.19.2-1.0.0
		techreborn: Tech Reborn 5.3.4
			reborncore: Reborn Core 5.3.4
		tismatis-roads: Tismatis's Roads 1.0.7
		upc: UPC 0.0.3
		viewdistancefix: View Distance Fix 1.19.2-1.0.0
		voicechat: Simple Voice Chat 1.19.2-2.3.11
		worldedit: WorldEdit 7.2.12+6240-87f4ae1
		xaerominimap: Xaero's Minimap 22.13.0
		xaeroworldmap: Xaero's World Map 1.26.2
	Server Running: true
	Player Count: 1 / 20; [class_3222['samuelh2005'/253, l='ServerLevel[City]', x=99.08, y=67.22, z=84.97]]
	Data Packs: vanilla, Fabric Mods, global:afk display v1.1.3, global:anti enderman grief v1.1.3, global:armor statues v2.9.2 (incompatible), global:camera (incompatible), global:cave cleaner (incompatible), global:concrete dyeing v1.2.0 (incompatible), global:coordinates hud v1.2.3, global:custom roleplay data v1.0 (incompatible), global:double shulker shells v1.3.3, global:durability ping v1.1.3, global:hard on start v1.0.0 (incompatible), global:more mob heads v2.10.1, global:multiplayer sleep v2.6.3, global:name_formatting_station_v2.0, global:nether portal coords v1.1.3, global:player head drops v1.1.3, global:silence mobs v1.1.3, global:spectator conduit power v1.1.3, global:spectator night vision v1.1.3, global:tp tools (incompatible), global:track raw statistics v1.5.3, global:track statistics v1.1.4, global:universal dyeing v1.2.1 (incompatible), global:unlock recipes trigger (incompatible), global:wandering trades v1.6.0
	World Generation: Stable
	Is Modded: Definitely; Server brand changed to 'fabric'
	Type: Dedicated Server (map_server.txt)
```

**Comment by ajh123** [comment]
The bug sill priests when Blaze Burner is removed from inventory, the main post is updated.

**Comment by ajh123** [comment]
Also when the heat source is updated in general

**Comment by ajh123** [comment]
Everything works fine when using WorldEdit to change the heat source

**Comment by waddon1** [comment]
I'm also experiencing this crash

**Comment by ground-fruit** [comment]
I am also experiencing this crash. Note that it also happens when adding a whistle to a tank.

**Comment by ajh123** [comment]
@mrh0 I know you know fabric, but is there anything you could help us with?

**Comment by ajh123** [comment]
Have anyone tried this on forge?

**Comment by toonystank** [comment]
I got the same issue in fabric

**Comment by waddon1** [comment]
> Have anyone tried this on forge?

Doesn't happen on forge

**Comment by ajh123** [comment]
#419 contains a solution but the developers need to see it first!

**Comment by mrh0** [comment]
Fixes in 20221219a.
